netdisco/discoverables/__init__.py

```diff
@@ -36,8 +36,15 @@ def __init__(self, netdis):
 
     def get_info(self):
         """Get most important info, by default the description location."""
-        return list(set(
-            self.info_from_entry(entry) for entry in self.get_entries()))
+        entries = self.get_entries()
+
+        # return list allowing for entries with both legacy tuples as well as
+        # later dictionaries
+        return list(
+            set(self.info_from_entry(entry)) if isinstance(entry, tuple)
```

Comment:
We only wrapped the whole result set in a set so that we wouldn't have duplicate entries. Kinda feels weird now so let's just remove it .

Actually, why would you change this logic at all?